### PR TITLE
Fix: UI Issue with Sidebar Submenu Overlapping When Sidebar is Pinned

### PR DESCRIPTION
**Pull request by mahendra1290**
## Hey, I just made a Pull Request!

Fixed a bug where the submenu in the sidebar would overlap with the pinned sidebar when hovering over the any grouped tab from sidebar.

Along with that added fade-in animation to the submenu which looks smooth.

closes: #25888

## Before

https://github.com/user-attachments/assets/7700d7e4-df99-438e-b4be-2871e260cda7

## After

https://github.com/user-attachments/assets/90a56100-c575-4679-97e8-7f1b380ddeac

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mahendra1290** [comment]
@Rugvip Can you please review this whenever you get time 😃

**Review by camilaibs — APPROVED** [review]
Thanks for fixing it, I left a minor changeset suggestion.
I would also appreciate a second review from @backstage/maintainers 🙂

File: .changeset/honest-impalas-rescue.md
Comment (mahendra1290) [reply .changeset/honest-impalas-rescue.md]: done

**Comment by mahendra1290** [comment]
@camilaibs a test failed randomly how I can re run tests again?

**Comment by camilaibs** [comment]
> @camilaibs a test failed randomly how I can re run tests again?

I think it is passing now

**Review by mkaliszewski — APPROVED** [review]

**Comment by nia-potato** [comment]
hi @Rugvip @camilaibs if everything looks good,  can we merge this?

**Review by Rugvip — APPROVED** [review]
Yep looks good, thank you, let's :shipit: :tada:

**Pull request merged**
